### PR TITLE
Add zstd compression check in SDK capabilities

### PR DESCRIPTION
**Pull request by rubensf**

### CODE REVIEW

File: go/pkg/client/capabilities.go [hunk 1]
```diff
@@ -28,6 +28,18 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
...
+	if c.serverCaps.CacheCapabilities.SupportedCompressors != nil && c.CompressedBytestreamThreshold >= 0  {
+		foundZstd := false
+		for _, sComp := range c.serverCaps.CacheCapabilities.SupportedCompressors {
+			if sComp == repb.Compressor_ZSTD {
+				foundZstd = true
+				break
+			}
+		}
+		if !foundZstd {
+			return errors.New("zstd compression is not supported by server")
```
Comment (bozydarsz) [inline go/pkg/client/capabilities.go:40]: should it be returned also when `CompressedBytestreamThreshold >= 0` and `SupportedCompressors = nil`?

**Review by bozydarsz — COMMENTED** [review]
it might be worth updating the comment of `CheckCapabilities` function to indicate that it checks whether zstd compressor is supported if the compression is enabled in client config

File: go/pkg/client/capabilities.go [hunk 2]
```diff
@@ -28,6 +28,18 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
...
 	}
+	if c.serverCaps.CacheCapabilities.SupportedCompressors != nil && c.CompressedBytestreamThreshold >= 0  {
+		foundZstd := false
+		for _, sComp := range c.serverCaps.CacheCapabilities.SupportedCompressors {
+			if sComp == repb.Compressor_ZSTD {
+				foundZstd = true
+				break
+			}
+		}
+		if !foundZstd {
```
[the remark this answers is not in the document]
Comment (rubensf) [reply go/pkg/client/capabilities.go:44]: Updated the error message - zstd is the only compression supported by the SDK (technically the only other one that the API supports is zip, but we haven't had a need to add support: https://github.com/bazelbuild/remote-apis/blob/main/build/bazel/remote/execution/v2/remote_execution.proto#L1735).

File: go/pkg/client/capabilities.go [hunk 3]
```diff
@@ -28,6 +28,18 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
 	if c.serverCaps.CacheCapabilities != nil {
 		c.MaxBatchSize = MaxBatchSize(c.serverCaps.CacheCapabilities.MaxBatchTotalSizeBytes)
 	}
+	if c.serverCaps.CacheCapabilities.SupportedCompressors != nil && c.CompressedBytestreamThreshold >= 0  {
```
[the remark this answers is not in the document]
Comment (rubensf) [reply go/pkg/client/capabilities.go]: Oh, good point! I updated to add a `useCompression` variable. PTAL

File: go/pkg/client/capabilities.go (hunk 1 above)
Comment (rubensf) [reply go/pkg/client/capabilities.go]: Yup -fixed.

File: go/pkg/client/capabilities.go [hunk 4]
```diff
@@ -28,6 +28,23 @@ func (c *Client) CheckCapabilities(ctx context.Context) (err error) {
 	if c.serverCaps.CacheCapabilities != nil {
 		c.MaxBatchSize = MaxBatchSize(c.serverCaps.CacheCapabilities.MaxBatchTotalSizeBytes)
 	}
+
+	if useCompression := c.CompressedBytestreamThreshold >= 0; useCompression {
```
Comment (gkousik) [inline go/pkg/client/capabilities.go:32]: nit: `compressionSupported` as variable name?

**Review by gkousik — APPROVED** [review]

File: go/pkg/client/capabilities.go (hunk 4 above)
Comment (rubensf) [reply go/pkg/client/capabilities.go:32]: Technically, this is whether the _user_ requested to use Compression. 

The next few checks are the ones checking whether compression is supported :)

Comment (gkousik) [reply go/pkg/client/capabilities.go:32]: Ah ok, then there isn't a need to use a variable here to explicitly identify it (it was a user specified option afterall). But tbh, I'm just ambivalent about it, so lgtm!

Comment (rubensf) [reply go/pkg/client/capabilities.go:32]: it's purely for readability (particularly if someone is dropped in midst of the code base) - see bozydar's comment above

Comment (gkousik) [reply go/pkg/client/capabilities.go:32]: Yep I saw that comment (which was what I disagreed with). I think the condition `c.CompressedBytestreamThreshold >= 0` is in itself very straightforward in stating that this block will be executed only if we are going to be using compression (there's very little room for misinterpretation).
Using a variable also seemed like a misdirection to me (I was searching for where this variable was being used again) - but again, this is all very subjective, which is why I didn't want to emphasize on it :)

**Pull request merged**